File: backend/app/schemas/report.py
from datetime import date
from pydantic import BaseModel

class ReportByStaffRow(BaseModel):
    staff_id: str
    total_tasks: int
    open_tasks: int
    closed_tasks: int

class ReportByTagRow(BaseModel):
    tag: str
    total_tasks: int

class ReportFilters(BaseModel):
    start_date: date | None = None
    end_date: date | None = None

File: src/types/report.ts
export type ReportsFilters = {
  start_date: string | null;
  end_date: string | null;
};

export type TasksSummaryStatusRow = { label: string; count: number };

export type TasksSummaryResponse = {
  generated_at: string;
  filters: ReportsFilters;
  total_tasks: number;
  open_tasks: number;
  closed_tasks: number;
  by_status: TasksSummaryStatusRow[];
};

export type StaffSummaryRow = {
  staff_id: string;
  total_tasks: number;
  open_tasks: number;
  closed_tasks: number;
};

export type StaffSummaryResponse = {
  generated_at: string;
  filters: ReportsFilters;
  items: StaffSummaryRow[];
};

export type TagSummaryRow = {
  tag: string;
  total_tasks: number;
};

export type TagSummaryResponse = {
  generated_at: string;
  filters: ReportsFilters;
  items: TagSummaryRow[];
};
